src/Microsoft.Graph.Cli.Core/Authentication/ClientCertificateCredentialFactory.cs

```diff
@@ -74,19 +76,26 @@ internal static bool TryGetCertificateFromStore(string certificateNameOrThumbPri
         try
         {
             store.Open(OpenFlags.ReadOnly);
-
-            // If using a certificate with a trusted root you do not need to FindByTimeValid, instead:
-            // currentCerts.Find(X509FindType.FindBySubjectDistinguishedName, certName, true);
-            X509Certificate2Collection signingCerts = store.Certificates.Find(X509FindType.FindByTimeValid, DateTime.Now, false)
-                .Find(isThumbPrint ? X509FindType.FindByThumbprint : X509FindType.FindBySubjectDistinguishedName, certificateNameOrThumbPrint, false);
+            X509Certificate2Collection signingCerts = store.Certificates
+                .Find(X509FindType.FindByTimeValid, DateTime.Now, true)
```

Comment:
I'm not sure why we're looking only at valid certificates? By definition self-signed certificates are "invalid", and that's what is mostly used in client credentials scenarios